### PR TITLE
Added platfrom validation testcases

### PR DESCRIPTION
**Pull request by bkhadars**

### CODE REVIEW

File: pkg/types/powervs/validation/platform.go [hunk 1]
```diff
@@ -6,17 +6,56 @@ import (
...
+var regionShortNames = func() []string {
+	keys := make([]string, len(powerVSRegions))
+	i := 0
+	for r := range powerVSRegions {
+		keys[i] = r
+		i++
+	}
+	return keys
+}()
+
```
Comment (clnperez) [inline pkg/types/powervs/validation/platform.go:35]: i'd prefer this to be a bit simpler if all you need is a list of the names:

```@@ -6,23 +6,12 @@ import (
        "github.com/openshift/installer/pkg/types/powervs"
 )
 
-var powerVSRegions = func() map[string]string {
-       regions := make(map[string]string)
-
-       for _, region := range powervs.Regions {
-               regions[region.Name] = regions[region.Description]
-       }
-       return regions
-}()
-
 var regionShortNames = func() []string {
-       keys := make([]string, len(powerVSRegions))
-       i := 0
-       for r := range powerVSRegions {
-               keys[i] = r
-               i++
+       var shortnames []string
+       for name := range powervs.Regions {
+               shortnames = append(shortnames, name)
        }
-       return keys
+       return shortnames
 }()
 
 // ValidatePlatform checks that the specified platform is valid.
@@ -31,7 +20,7 @@ func ValidatePlatform(p *powervs.Platform, fldPath *field.Path) field.ErrorList
 
        if p.Region == "" {
                allErrs = append(allErrs, field.Required(fldPath.Child("region"), "region must be specified"))
-       } else if _, ok := powerVSRegions[p.Region]; !ok {
+       } else if _, ok := powervs.Regions[p.Region]; !ok {
                allErrs = append(allErrs, field.NotSupported(fldPath.Child("region"), p.Region, regionShortNames))
        }
```

File: pkg/types/powervs/validation/platform.go [hunk 2]
```diff
@@ -6,17 +6,56 @@ import (
...
+	allErrs := field.ErrorList{}
+	if p.VPC != "" || len(p.Subnets) > 0 || p.PowerVSResourceGroup != "" {
+		if p.VPC == "" {
+			allErrs = append(allErrs, field.Required(path.Child("vpc"), "vpc is required when specifying subnets or ResourceGroup"))
+		}
+		if len(p.Subnets) == 0 {
+			allErrs = append(allErrs, field.Required(path.Child("subnets"), "subnets is required when specifying vpc or ResourceGroup"))
+		}
+		if p.PowerVSResourceGroup == "" {
+			allErrs = append(allErrs, field.Required(path.Child("ResourceGroup"), "ResourceGroup is required when specifying vpc or subnets"))
```
Comment (clnperez) [inline pkg/types/powervs/validation/platform.go]: it's required for vpc, etc., but I don't think it's important that users know what components its used to create.

File: pkg/types/powervs/validation/platform_test.go [hunk 3]
```diff
@@ -3,174 +3,170 @@ package validation
...
 		{
-			name: "valid url for service endpoint",
-			platform: &aws.Platform{
-				Region: "us-east-1",
-				ServiceEndpoints: []aws.ServiceEndpoint{{
-					Name: "ec2",
-					URL:  "test-ec2.random.local",
-				}},
-			},
+			name: "invalid vpc config missing vpc",
```
Comment (clnperez) [inline pkg/types/powervs/validation/platform_test.go]: for these names, can you add some punctuation? "invalid vpc: config missing vpc" or something like that?

File: pkg/types/powervs/validation/platform_test.go [hunk 4]
```diff
@@ -3,174 +3,170 @@ package validation
...
 		{
-			name: "valid url for service endpoint",
-			platform: &aws.Platform{
-				Region: "us-east-1",
-				ServiceEndpoints: []aws.ServiceEndpoint{{
-					Name: "ec2",
-					URL:  "https://test-ec2.random.local",
-				}},
-			},
+			name: "invalid vpc config missing subnets",
```
Comment (clnperez) [inline pkg/types/powervs/validation/platform_test.go]: same here, etc.

**Comment by clnperez** [comment]
a few nits but looks good overall. i may hold off on merging this until we get the initial prs submitted so they're a bit smaller.

**Comment by clnperez** [comment]
ping @bkhadars

File: pkg/types/powervs/validation/platform_test.go (hunk 4 above)
Comment (bkhadars) [reply pkg/types/powervs/validation/platform_test.go]: fixed

File: pkg/types/powervs/validation/platform_test.go (hunk 3 above)
Comment (bkhadars) [reply pkg/types/powervs/validation/platform_test.go]: fixed

File: pkg/types/powervs/validation/platform.go (hunk 1 above)
Comment (bkhadars) [reply pkg/types/powervs/validation/platform.go:35]: fixed

**Comment by bkhadars** [comment]
@clnperez Pushed the latest changes

**Comment by clnperez** [comment]
@bkhadars where did you push these? am i forgetting a PR?

**Pull request closed without merging**